ranger/ext/vcs/hg.py

```diff
@@ -1,271 +1,162 @@
-# -*- coding: utf-8 -*-
 # This file is part of ranger, the console file manager.
 # License: GNU GPL version 3, see the file "AUTHORS" for details.
-# Author: Abdó Roig-Maranges <abdo.roig@gmail.com>, 2011-2012
-#
-# vcs - a python module to handle various version control systems
 
-import os
-import re
-import shutil
+"""Mercurial module"""
+
 from datetime import datetime
-try:
-    from configparser import RawConfigParser
-except ImportError:
-    from ConfigParser import RawConfigParser
+import json
+import os
 
 from .vcs import Vcs, VcsError
 
 
 class Hg(Vcs):
-    vcsname  = 'hg'
+    """VCS implementation for Mercurial"""
     HEAD = 'tip'
-    # Auxiliar stuff
-    #---------------------------
-
-    def _hg(self, path, args, silent=True, catchout=False, bytes=False):
-        return self._vcs(path, 'hg', args, silent=silent, catchout=catchout, bytes=bytes)
-
-
-    def _has_head(self):
-        """Checks whether repo has head"""
-        rnum = self._hg(self.path, ['-q', 'identify', '--num', '-r', self.HEAD], catchout=True)
-        return rnum != '-1'
-
-
-    def _sanitize_rev(self, rev):
-        if rev == None: return None
-        rev = rev.strip()
-        if len(rev) == 0: return None
-        if rev[-1] == '+': rev = rev[:-1]
 
-        try:
-            if int(rev) == 0: return None
-        except:
-            pass
+    _status_translations = (
+        ('AR', 'staged'),
+        ('M', 'changed'),
+        ('!', 'deleted'),
 
-        return rev
+        ('?', 'untracked'),
+        ('I', 'ignored'),
+    )
 
+    # Generic
 
     def _log(self, refspec=None, maxres=None, filelist=None):
 
-        fmt = "changeset: {rev}:{node}\ntag: {tags}\nuser: {author}\ndate: {date}\nsummary: {desc}\n"
-        args = ['log', '--template', fmt]
-
-        if refspec:  args = args + ['--limit', '1', '-r', refspec]
-        elif maxres: args = args + ['--limit', str(maxres)]
+        args = [
+            'log', '--template',
+            '\\{'
+            '\\x00short\\x00:\\x00{rev}\\x00,'
+            '\\x00revid\\x00:\\x00{node}\\x00,'
+            '\\x00author\\x00:\\x00{author}\\x00,'
+            '\\x00date\\x00:\\x00{date}\\x00,'
+            '\\x00summary\\x00:\\x00{desc}\\x00'
+            '}\\n'
+        ]
```

Comment:
Let's use `--template json` instead of this, it will be more maintainable and less error-prone,